src/zod.ts

```diff
@@ -0,0 +1,95 @@
+import {ValidationResult} from "./types"
+import {generateErrorMessage} from 'zod-error'
+
+class _Zod {
+
+  public readonly metadata: any
+  private z: any
+
+  private constructor(metadata: any) {
+    this.metadata = metadata
+  }
+
+  static async init(): Promise<_Zod> {
+    const res = await fetch('https://data.jsdelivr.com/v1/packages/npm/zod')
+    const metadata = await res.json()
+
+    // metadata.tags
+
+    // alpha: "3.21.5-alpha.0"
+    // beta: "3.23.0-beta.2"
+    // canary: "3.24.0-canary.20240523T174819"
+    // latest: "3.23.8"
+    // next: "3.8.2-alpha.6"
+
+    // 4.0.0-beta.1 in metadata.versions
+
+    return new _Zod(metadata)
+  }
+
+  async setVersion(ver: string) {
+    this.z = await import(/* @vite-ignore */ `https://cdn.jsdelivr.net/npm/zod@${ver}/+esm`)
```

Comment:
this is so nice, didn't know we could use full urls within dynamic imports

Reply:
```
1:39:46 PM [vite] warning: invalid import "../https:/cdn.jsdelivr.net/npm/zod@${ver}/+esm". A file extension must be included in the static part of the import. For example: import(`./foo/${bar}.js`).
  Plugin: vite:dynamic-import-vars
  File: /home/davide/zod-playground/src/zod.ts
1:39:46 PM [vite] warning: 
/home/davide/zod-playground/src/zod.ts
12 |    }
13 |    async setVersion(ver) {
14 |      this.z = await import(`https://cdn.jsdelivr.net/npm/zod@${ver}/+esm`);
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
15 |    }
16 |    validateSchema(schema) {
The above dynamic import cannot be analyzed by Vite.
See https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations for supported dynamic import formats. If this is intended to be left as-is, you can use the /* @vite-ignore */ comment inside the import() call to suppress this warning.

  Plugin: vite:import-analysis
  File: /home/davide/zod-playground/src/zod.ts
  ```